smart-py/marketplace.py

```diff
@@ -1,65 +1,954 @@
+        valid = False
+    )
+
+@sp.add_test("Test collect Multiple Editions")
+def test():
+    # init test and create html output
+    scenario = sp.test_scenario()
+    scenario.h1("Collect Test (Multiple Edition)")
+
+    # init test values
+    seller = sp.test_account("seller")
+    buyer = sp.test_account("buyer")
+    manager = sp.test_account("manager")
+    objkt = sp.test_account("objkt123")
+
+    creator = sp.test_account("creator")
+    metadata = sp.record(
+        name = "test",
+        description = "test",
+        tags = [
+            'test'
+        ],
+        symbol = 'OBJKT',
+        artifactUri = "ipfs://test",
+        displayUri = "ipfs://test",
+        thumbnailUri = "ipfs://test",
+        creators = [
+            creator.address
+        ],
+        formats = [
+            {
+                "uri":"ipfs://test",
+                "mimeType":"image/png"
+            }
+        ],
+        decimals = 0,
+        isBooleanAmount = False,
+        shouldPreferSymbol = False
+    )
+
+    fee = 25
+
+    swap = Marketplace(
+        objkt.address,
+        metadata,
+        manager.address,
+        fee
+    )
+
+    scenario += swap
+
+    # no swaps yet
+    scenario.verify(swap.data.counter == 500000)
+    scenario.verify(swap.data.swaps.contains(500000) == False)
+
+    scenario += swap.swap(
+        creator=creator.address,
+        royalties=250,
+        objkt_id = 153,
+        objkt_amount = 10,
+        xtz_per_objkt = sp.utils.nat_to_mutez(3)
+    ).run(
+        sender = seller.address,
+        valid = True
+    )
+
+    # one swap was added
+    scenario.verify(swap.data.swaps.contains(500000) == True)
+    scenario.verify(swap.data.swaps.contains(500001) == False)
+    scenario.verify(swap.data.counter == 500001)
+
+    # data is as expected inside the swap
+    scenario.verify(swap.data.swaps.get(500000).objkt_id == 153)
+    scenario.verify(swap.data.swaps.get(500000).objkt_amount == 10)
+    scenario.verify(swap.data.swaps.get(500000).royalties == 250)
+    scenario.verify(swap.data.swaps.get(500000).xtz_per_objkt == sp.utils.nat_to_mutez(3))
+
+    # collect with enough tez
+    scenario += swap.collect(
+        swap_id = 500000,
+    ).run(
+        sender = seller.address,
+        amount = sp.utils.nat_to_mutez(3),
+        valid = True
+    )
+
+    # the swap is still present but the available objkts is now 9
+    scenario.verify(swap.data.swaps.get(500000).objkt_id == 153)
+    scenario.verify(swap.data.swaps.get(500000).objkt_amount == 9)
+    scenario.verify(swap.data.swaps.get(500000).royalties == 250)
+    scenario.verify(swap.data.swaps.get(500000).xtz_per_objkt == sp.utils.nat_to_mutez(3))
+
+    # collect with enough tez
+    scenario += swap.collect(
+        swap_id = 500000,
+    ).run(
+        sender = seller.address,
+        amount = sp.utils.nat_to_mutez(3),
+        valid = True
+    )
+
+    # the swap is still present but the available objkts is now 8
```

Comment:
Should you also have a test when the available objkts goes down to 1 and 0 and collects are done?

Reply:
good point, i've expanded the tests to cover this